doc: make headings more visually identifiable

kartben opened the pull request:
https://builds.zephyrproject.io/zephyr/pr/77737/docs/develop/getting_started/index.html

Before this change, headings in documentation pages were using the same font family as the rest of the content, which made them blend in too much.

kartben committed:
doc: make headings more visually identifiable

Before this change, headings in documentation pages were using the same
font family as the rest of the content, which made them blend in too
much.

This introduces a humanist font stack for the headings as well as some
minor tweaks to make the hierarchy of the document clearer through
better differentiation of the different levels (i.e. not just relying on
font size).

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

pdgendt commented:
Can't say that I'm a fan of the headers with a serif font.

pdgendt commented:
> @pdgendt I am pretty sure I just opened a can of worms :) but in any case, do you happen to like this one better? It doesn't really make the headings stand out as much though

It all comes down to preference, or maybe some "marketing" specifications.
The latter sans-serif would get my vote though :)

stephanosio commented on doc/_static/css/custom.css:
```diff
@@ -7,6 +7,12 @@
+ :root {
+     /* Use system font stacks for better performance (no Web fonts required) */
+     --system-font-family: system-ui, -apple-system, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
+     --header-font-family: Charter, 'Bitstream Charter', 'Sitka Text', Cambria, serif;
```
I have to admit sans-serif looks more "clean" and "modern" ...

pdgendt commented:
+1 for the styling

kartben commented:
> +1 for the styling

Thanks for the feedback! Will likely go through a final tweak of the H1 and will put out of draft when ready

kartben committed:
doc: css: fix mobile layout and increase browser compatibility

added a missing dvh causing footer to be footer to be displayed outside
of the viewport on mobile when browsing "Desktop site" version, and also
added fallback "vh" rules for slightly better browser compatibility.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>
